Makefile

```diff
@@ -42,14 +42,33 @@ pod-status-check:
 workloads:
 	./scripts/workloads.sh
 
+./PHONY: workloads-spin-registry-push
+workloads-spin-registry-push:
+	./scripts/workloads-spin-registry-push.sh
+
 ./PHONY: test-workloads-delete
 test-workloads-delete:
 	./scripts/workloads-delete.sh
 
 .PHONY: integration-tests
-integration-tests: check-bins move-bins up pod-status-check workloads test-workloads-delete
+integration-tests: prepare-cluster-and-images integration-docker-build-push-tests integration-spin-registry-push-tests
+
+.PHONY: integration-docker-build-push-tests
+integration-docker-build-push-tests: workloads
+	cargo test -p containerd-shim-spin-tests -- --nocapture
```

Comment:
I think this will fail currently. do you think that is a fair outcome or you would expect it to fail these tests and run the other set of tests (spin-registry-push testcases)

given we are using same k3d cluster to run both tests right now, i think if it fails, we should fail the complete job. As suggested in one of the other comment on this PR, we should do a follow up PR to separate these two tasks as separate steps in CI workflow.

Reply:
I'm in support of failing the complete job if this task fails.  I was more worried about lingering resources when this `cargo test ...` command fails and so the subsequent `kubectl delete ...` cleanup commands don't run.  I suppose in CI it doesn't matter... but maybe in dev scenarios?

Reply:
Hi @vdice, how do you suggest we handle this? one way would be to capture the output and exit code and then throw a new error after cleanup?

Reply:
It looks like this logic is now in a shell script.  I'd say let's not worry about it for this PR and we can add a follow-up if we run into this scenario when developing/running tests locally.

Reply:
👍 , i moved it to shell script in response to other comments around unifying some of the scripts. agree that we can work on a separate PR to improve this further.